File: 242-valid-anagram/242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(s.length !== t.length){
        return false
    }
    let sArray = s.split('')
    let tArray = t.split('')
    let fqCounter1 = {}
    let fqCounter2 = {}
    
    for(let val of sArray){
        fqCounter1[val] = (fqCounter1[val] || 0) + 1;
    }
    for(let val of tArray){
        fqCounter2[val] = (fqCounter2[val] || 0) + 1;
    }
    
    for (let key in fqCounter1){
        if(!(key in fqCounter2)){
            return false;
        }
        if(fqCounter2[key] !== fqCounter1[key]){
            return false
        }
    }
    return true;
};

File: 242-valid-anagram/242-valid-anagram.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_anagram(s, t)
    return false unless s.length == t.length
    
    s_array = s.split('')
    t_array = t.split('')
    fq_counter1 = s_array.tally
    fq_counter2 = t_array.tally
    
   fq_counter1 == fq_counter2
end 
